crates/sui/src/client_commands.rs

```diff
@@ -607,6 +609,54 @@ pub enum SuiClientCommands {
         #[clap(long)]
         address_override: Option<ObjectID>,
     },
+
+    /// Replay a given transaction to view transaction effects. Set environment variable MOVE_VM_STEP=1 to debug.
+    #[clap(name = "replay-transaction")]
+    ReplayTransaction {
+        /// The rpc url for a non-pruning fullnode to use for fetching the transaction dependencies
+        #[arg(long = "rpc")]
+        rpc_url: String,
+
+        /// The digest of the transaction to replay
+        #[arg(long, short)]
+        tx_digest: String,
+    },
+
+    /// Replay transactions listed in a file.
+    #[clap(name = "replay-batch")]
+    ReplayBatch {
+        /// The rpc url for a non-pruning fullnode to use for fetching the transaction dependencies
+        #[arg(long = "rpc")]
+        rpc_url: String,
+
+        /// The path to the file of transaction digests to replay, with one digest per line
+        #[arg(long, short)]
+        path: PathBuf,
+
+        /// If an error is encountered during a transaction, this specifies whether to terminate or continue
+        #[arg(long, short)]
+        terminate_early: bool,
+    },
+
+    /// Replay all transactions in a range of checkpoints.
+    #[command(name = "replay-checkpoint")]
+    ReplayCheckpoints {
+        /// The rpc url for a non-pruning fullnode to use for fetching the transaction dependencies
+        #[arg(long = "rpc")]
+        rpc_url: String,
+
+        /// The start value of the range of checkpoints to replay
```

Comment:
nit: Can we be explicit about what unit/format of this argument is? I guess it's the checkpoint sequence number? I'm also not sure how pervasive that is as a unit used outside of our circles (e.g. whether people are used to checkpoints having a global order, or whether it makes sense to refer to these by the checkpoint digest).